src/metabase/models/field_usage.clj

```diff
@@ -0,0 +1,100 @@
+(ns metabase.models.field-usage
+  (:require
+   [metabase.lib.core :as lib]
+   [metabase.lib.dispatch :as lib.dispatch]
+   [metabase.lib.hierarchy :as lib.hierarchy]
+   [metabase.lib.schema :as lib.schema]
+   [metabase.lib.util :as lib.util]
+   [metabase.lib.util.match :as lib.util.match]
+   [metabase.models.interface :as mi]
+   [metabase.query-processor.middleware.fetch-source-query :as fetch-source-query]
+   [metabase.util.malli :as mu]
+   [methodical.core :as methodical]
+   [toucan2.core :as t2]
+   [toucan2.tools.disallow :as t2.disallow]))
+
+(doto :model/FieldUsage
+  (derive :metabase/model)
+  (derive ::t2.disallow/update)
+  (derive :hook/created-at-timestamped?))
+
+(methodical/defmethod t2/table-name :model/FieldUsage [_model] :field_usage)
+
+(t2/deftransforms :model/FieldUsage
+  {:used_in                mi/transform-keyword
+   :aggregation_function   mi/transform-keyword
+   :breakout_temporal_unit mi/transform-keyword
+   :breakout_binning       mi/transform-json
+   :filter_op              mi/transform-keyword
+   :filter_args            mi/transform-json})
+
+(defn- filter->field-usage
+  [query stage filter-clause]
+  (let [filter-parts (lib/filter-parts query stage filter-clause)
+        field-id     (-> filter-parts :column :id)]
+    (when (int? field-id)
+      {:field_id    field-id
+       :used_in     :filter
+       :filter_op   (-> filter-parts :operator :short)
```

Comment:
I guess reaching into `::lib.schema.filter/operator` is allowed?

Reply:
not sure, but we surely don't want to store the whole map. All the filter functions return this operator map :(.

Reply:
yeah, for sure. Would be nice to get qpd eyes on this though.

Reply:
`lib/filter-parts` and `lib/expression-parts` return those operator maps, which are intended to be an external AST for expressions and a public interface. So **yes**, feel free to look into those maps and pull out just the pieces you need, like in this code.